simplecpp.cpp

```diff
@@ -870,7 +878,8 @@ std::string simplecpp::TokenList::lastLine(int maxsize) const {
             continue;
         if (!ret.empty())
             ret = ' ' + ret;
-        ret = (tok->str[0] == '\"' ? std::string("%str%") : tok->str) + ret;
+        ret = (tok->str[0] == '\"' ? std::string("%str%")
+               : isdigit(tok->str[0]) ? std::string("%num%") : tok->str) + ret;
```

Comment:
likewise for isdigit. is it _known_ that str[0] is 0-0x7f here? or should there be a cast to unsigned char so isdigit will only get 0x00-0xff values.